### PR TITLE
Add zip deploy

### LINKED ISSUE #3: Deploy function app as zip package to Azure

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
This leverages the shared package I've created here: https://github.com/EricJizbaMSFT/vscode-azure-sdk

Obviously we'll have to discuss if that shared package will work and find an actual home for it instead of my fork.

Fixes #3

### CODE REVIEW

File: src/commands/deployZip.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const newFolderId: string = 'newFolder';
+    let folderPicks: util.PickWithData<string>[] = [new util.PickWithData(newFolderId, localize('azFunc.selectOtherFolder', '$(plus) Select other folder'))];
+    const folders: vscode.WorkspaceFolder[] | undefined = vscode.workspace.workspaceFolders;
+    if (folders) {
+        folderPicks = folderPicks.concat(folders.map((f: vscode.WorkspaceFolder) => new util.PickWithData<string>('', f.uri.fsPath)));
+    }
+    const folder: util.PickWithData<string> = await util.showQuickPick<string>(folderPicks, localize('azFunc.deployZipSelectFolder', 'Select a workspace folder to deploy to your Function App'));
+    const folderPath: string = folder.data === newFolderId ? await util.showFolderDialog() : folder.label;
+
+    await node.siteWrapper.deployZip(folderPath, client, outputChannel);
```
Comment (StephenWeatherford) [inline src/commands/deployZip.ts]: outputChannel.show();

Should also add it to create function app, etc.

Comment (EricJizbaMSFT) [reply src/commands/deployZip.ts]: Added this inside of siteWrapper.deployZip

File: package.nls.json [hunk 2]
```diff
@@ -7,5 +7,6 @@
     "azFunc.startFunctionApp": "Start",
     "azFunc.stopFunctionApp": "Stop",
     "azFunc.restartFunctionApp": "Restart",
-    "azFunc.showExplorerDescription": "Show or hide the Azure Functions Explorer"
+    "azFunc.showExplorerDescription": "Show or hide the Azure Functions Explorer",
+    "azFunc.deployZip": "Deploy as Zip Package"
```
Comment (StephenWeatherford) [inline package.nls.json]: How about "Deploy here as zip package"?

Comment (StephenWeatherford) [reply package.nls.json]: (but capitalized properly)

Comment (EricJizbaMSFT) [reply package.nls.json]: But 'here' doesn't make sense when the command is run from the command palette.

I created a bug to address the entry points separately: https://github.com/Microsoft/vscode-azurefunctions/issues/15

**Comment by EricJizbaMSFT** [comment]
Okay the PR for the shared packages is out: https://github.com/Microsoft/vscode-azuretools/pull/1

This PR is updated with latest from master and I removed link to package in my custom fork. CI build will fail until shared package is published.

**Pull request merged**